### PR TITLE
Adds build and configure scripts

### PR DESCRIPTION
**Pull request by KipHamiltons**
Adds the first `b` tools for NUWebots, with `./b build` and `./b configure`.
Depends on #39

### CODE REVIEW

File: scripts/configure.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+
+def run(interactive, args, **kwargs):
+
+    # Make sure we have a build directory
+    try:
+        os.makedirs("build")
+    except FileExistsError:
+        # directory already exists
+        pass
```
Comment (Bidski) [inline scripts/configure.py:31]:
```suggestion
    os.makedirs("build", exist_ok=True)
```

File: scripts/configure.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    os.chdir(os.path.join(b.project_dir, "build"))
+
+    command = ["ccmake" if interactive else "cmake"]
+
+    # To pass arguments to the cmake command you put them after "--"
+    # but "--"  isn't a valid argument for cmake, so we remove it here
+    if "--" in args:
+        args.remove("--")
+
+    command.extend([*args, b.project_dir])
```
Comment (Bidski) [inline scripts/configure.py:43]: I feel like we should make people use `ninja` as it is, overall, the better build tool
```suggestion
    command.extend([*args, b.project_dir, "-GNinja"])
```

File: scripts/build.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    command.add_argument("-j", help="number of jobs to spawn")
+
+
+def run(j, args, **kwargs):
+    # Change into the build directory
+    os.chdir(os.path.join(b.project_dir, "build"))
+
+    # Run cmake if we don't have a ninja build file
+    if not os.path.isfile("Makefile"):
+        exitcode = os.system("cmake {}".format(b.project_dir)) >> 8
```
Comment (Bidski) [inline scripts/build.py:26]:
```suggestion
        exitcode = os.system("cmake {} -GNinja".format(b.project_dir)) >> 8
```

File: scripts/build.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        if exitcode != 0:
+            exit(exitcode)
+
+    # Default to using all cores
+    if j is None:
+        j = cpu_count()
+
+    command = ["make", *args]
+
+    command.insert(1, "-j{}".format(j))
```
Comment (Bidski) [inline scripts/build.py:38]:
```suggestion
    command = ["ninja", "-j{}".format(j), *args]
```

File: scripts/build.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    command.add_argument("args", nargs="...", help="the arguments to pass through to ninja")
+    command.add_argument("-j", help="number of jobs to spawn")
+
+
+def run(j, args, **kwargs):
+    # Change into the build directory
+    os.chdir(os.path.join(b.project_dir, "build"))
+
+    # Run cmake if we don't have a ninja build file
+    if not os.path.isfile("Makefile"):
```
Comment (Bidski) [inline scripts/build.py:25]:
```suggestion
    if not os.path.isfile("build.ninja"):
```

**Review by Bidski — CHANGES_REQUESTED** [review]

**Comment by KipHamiltons** [comment]
We will roll with ninja, as per your suggestion @Bidski. I'll make a PR to NUbook

**Review by Bidski — APPROVED** [review]
LGTM

**Pull request merged**